src/server/game/Movement/MovementPacketSender.cpp

```diff
@@ -0,0 +1,106 @@
+    { SMSG_SPLINE_SET_RUN_BACK_SPEED,    SMSG_FORCE_RUN_BACK_SPEED_CHANGE,       MSG_MOVE_SET_RUN_BACK_SPEED },
+    { SMSG_SPLINE_SET_SWIM_SPEED,        SMSG_FORCE_SWIM_SPEED_CHANGE,           MSG_MOVE_SET_SWIM_SPEED },
+    { SMSG_SPLINE_SET_SWIM_BACK_SPEED,   SMSG_FORCE_SWIM_BACK_SPEED_CHANGE,      MSG_MOVE_SET_SWIM_BACK_SPEED },
+    { SMSG_SPLINE_SET_TURN_RATE,         SMSG_FORCE_TURN_RATE_CHANGE,            MSG_MOVE_SET_TURN_RATE },
+    { SMSG_SPLINE_SET_FLIGHT_SPEED,      SMSG_FORCE_FLIGHT_SPEED_CHANGE,         MSG_MOVE_SET_FLIGHT_SPEED },
+    { SMSG_SPLINE_SET_FLIGHT_BACK_SPEED, SMSG_FORCE_FLIGHT_BACK_SPEED_CHANGE,    MSG_MOVE_SET_FLIGHT_BACK_SPEED },
+    { SMSG_SPLINE_SET_PITCH_RATE,        SMSG_FORCE_PITCH_RATE_CHANGE,           MSG_MOVE_SET_PITCH_RATE },
+};
+
+MovementChangeType MovementPacketSender::GetChangeTypeByMoveType(UnitMoveType moveType)
+{
+    switch (moveType)
+    {
+        case MOVE_WALK:         return MovementChangeType::SPEED_CHANGE_WALK;
+        case MOVE_RUN:          return MovementChangeType::SPEED_CHANGE_RUN;
+        case MOVE_RUN_BACK:     return MovementChangeType::SPEED_CHANGE_RUN_BACK;
+        case MOVE_SWIM:         return MovementChangeType::SPEED_CHANGE_SWIM;
+        case MOVE_SWIM_BACK:    return MovementChangeType::SPEED_CHANGE_SWIM_BACK;
+        case MOVE_TURN_RATE:    return MovementChangeType::RATE_CHANGE_TURN;
+        case MOVE_FLIGHT:       return MovementChangeType::SPEED_CHANGE_FLIGHT_SPEED;
+        case MOVE_FLIGHT_BACK:  return MovementChangeType::SPEED_CHANGE_FLIGHT_BACK_SPEED;
+        case MOVE_PITCH_RATE:   return MovementChangeType::RATE_CHANGE_PITCH;
+        default:
+            ASSERT(false, "MovementPacketSender::SendSpeedChangeToMover Unsupported UnitMoveType");
+    }
+}
+
+void MovementPacketSender::SendSpeedChangeToMover(Unit* unit, UnitMoveType mtype, float newSpeedFlat)
+{
+    GameClient* controller = unit->GetGameClientMovingMe();
+
+    ASSERT(controller);
+
+    uint32 mCounter = unit->GetMovementCounterAndInc();
+    PlayerMovementPendingChange pendingChange;
+    pendingChange.movementCounter = mCounter;
+    pendingChange.newValue = newSpeedFlat;
+    pendingChange.movementChangeType = MovementPacketSender::GetChangeTypeByMoveType(mtype);
+    unit->PushPendingMovementChange(pendingChange);
+
+    WorldPacket data;
+    data.Initialize(moveTypeToOpcode[mtype][1], mtype != MOVE_RUN ? 8 + 4 + 4 : 8 + 4 + 1 + 4);
+    data << unit->GetPackGUID();
+    data << mCounter;
+    if (mtype == MOVE_RUN)
+        data << uint8(1);                               // unknown byte added in 2.1.0
+    data << newSpeedFlat;
+    controller->SendDirectMessage(&data);
+}
+
+void MovementPacketSender::SendSpeedChangeToObservers(Unit* unit, UnitMoveType mtype, float newSpeedFlat)
+{
+    GameClient* controller = unit->GetGameClientMovingMe();
+
+    ASSERT(controller);
+
+    WorldPacket data;
```

Comment:
maybe this is a nice chance to move it to the new packet format, if you feel like it

Reply:
yeah please do so :)

Reply:
Yeah, but that's another PR.

Reply:
"if you feel like it"

Reply:
Nah, since we're using array based packet building here, having a structure build helper here is cleaner that squeezing it all into one class imo. Also out of scope of this PR

Reply:
so won't do here (but everyone is welcome to improve this in a later PR)